js/apps/admin-ui/src/clients/credentials/SignedJWT.tsx

```diff
@@ -13,11 +13,15 @@ import { HelpItem } from "ui-shared";
 import { convertAttributeNameToForm, sortProviders } from "../../util";
 import { FormFields } from "../ClientDetails";
 
-export const SignedJWT = () => {
+type SignedJWTProps = {
+  clientAuthenticatorType: string;
+};
+
+export const SignedJWT = ({ clientAuthenticatorType }: SignedJWTProps) => {
   const { control } = useFormContext();
-  const providers = sortProviders(
-    useServerInfo().providers!.clientSignature.providers
-  );
+  const providers =  clientAuthenticatorType === "client-jwt"
+    ? (useServerInfo().cryptoInfo?.clientSignatureAsymmetricAlgorithms ?? [])
+    : (useServerInfo().cryptoInfo?.clientSignatureSymmetricAlgorithms ?? []);
```

Comment:
Although this does not result in errors, hooks [should not be called conditionally](https://react.dev/warnings/invalid-hook-call-warning). Instead you can store the value in side of a variable outside of these conditions:

```suggestion
  const { cryptoInfo } = useServerInfo();
  const providers =
    clientAuthenticatorType === "client-jwt"
      ? cryptoInfo?.clientSignatureAsymmetricAlgorithms ?? []
      : cryptoInfo?.clientSignatureSymmetricAlgorithms ?? [];`
``